test/libevmasm/Optimiser.cpp

```diff
@@ -1346,7 +1347,7 @@ BOOST_AUTO_TEST_CASE(jumpdest_removal_subassemblies)
 	settings.evmVersion = solidity::test::CommonOptions::get().evmVersion();
 	settings.expectedExecutionsPerDeployment = OptimiserSettings{}.expectedExecutionsPerDeployment;
 
-	Assembly main{settings.evmVersion, false, solidity::test::CommonOptions::get().eofVersion(), {}};
+	Assembly main{settings.evmVersion, false, std::nullopt, {}};
 	AssemblyPointer sub = std::make_shared<Assembly>(settings.evmVersion, true, solidity::test::CommonOptions::get().eofVersion(), std::string{});
```

Comment:
Why remove the EOF flag from the main assembly but still leave it in the subassembly?

In any case, I'd just leave the flag as it was. The test won't run on EOF anyway and if you remove the value I'd recommend asserting that `eofVersion()` is `nullopt` anyway. Leaving it in we'll get that assert for free - from anything inside `Assembly` that does not support EOF.

Reply:
Done.